lib/factorial.rb

```diff
@@ -1,4 +1,18 @@
 # Computes factorial of the input number and returns it
 def factorial(number)
-  raise NotImplementedError
+  raise ArgumentError if !number
+  return 1 if number == 1 || number == 0
+  
+  count = number
+  min = 1
```

Comment:
As additional challenge for code cleanup, check if you can do with just _count_ or _min_ and get rid of the other.